plugins/chat/assets/javascripts/discourse/services/chat-channel-thread-indicator-subscriptions-manager.js

```diff
@@ -0,0 +1,56 @@
+import Service, { inject as service } from "@ember/service";
+import { bind } from "discourse-common/utils/decorators";
+
+export default class ChatChannelThreadIndicatorSubscriptionsManager extends Service {
+  @service chat;
+  @service currentUser;
+
+  get messageBusChannel() {
+    return `/chat/${this.model.id}`;
+  }
+
+  get messageBusLastId() {
+    return this.model.channelMessageBusLastId;
+  }
+
+  get messagesManager() {
+    return this.model.messagesManager;
+  }
+
+  subscribe(model) {
+    this.unsubscribe();
+    this.model = model;
+    this.messageBus.subscribe(
+      this.messageBusChannel,
+      this.onMessage,
+      this.messageBusLastId
+    );
+  }
+
+  unsubscribe() {
+    if (!this.model) {
+      return;
+    }
+    this.messageBus.unsubscribe(this.messageBusChannel, this.onMessage);
+    this.model = null;
+  }
+
+  @bind
+  onMessage(busData) {
+    switch (busData.type) {
+      case "update_thread_indicator":
+        this.handleThreadIndicatorUpdate(busData);
+        break;
+    }
+  }
+
+  handleThreadIndicatorUpdate(data) {
+    const message = this.messagesManager.findMessage(data.original_message_id);
+    if (message) {
+      if (data.action === "increment_reply_count") {
+        // TODO (martin) In future we should use a replies_count delivered from the server.
+        message.threadReplyCount += 1;
+      }
+    }
+  }
+}
```

Comment:
We are not doing a `handleThreadIndicatorUpdate` here. We are updating the reply count of the original message of a thread. The consequences on the UI will happen later.

Reply:
Yes I see your meaning, I will change the wording here and in message bus to be around updating the original message. The thread indicators in the UI will potentially have a bunch of stuff happening to them, but you are right that they just represent the OMs.

Reply:
For now I am leaving this "action" as is though, since I don't have accurate reply counts to send from the server to the frontend. All we want to do is make sure the frontend increments for now then we can make it more accurate later.